File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int n = nums.Length;
        Dictionary<int, int> ump = new Dictionary<int, int>();
        int[] result = new int[2];

        for (int i = 0; i < n; i++) {
            int complement = target - nums[i];
            if (ump.ContainsKey(complement) && ump[complement] != i) {
                result[0] = ump[complement];
                result[1] = i;
                return result;
            }
            if (!ump.ContainsKey(nums[i])) {
                ump[nums[i]] = i;
            }
        }
        return null;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int,int>ump;
        vector<int>v;
        for(int i=0; i<n; i++)
        {
            int complement = target - nums[i];
            if(ump.find(complement)!=ump.end() && ump[complement]!=i)
            {
                auto p = ump.find(complement);
                v.push_back(p->second);
                v.push_back(i);
                break;
            }
            ump[nums[i]] = i;
        }
        return v;
    }
};